### ISSUE TITLE
Homepage: Information on footprint could be misleading

### ISSUE THREAD

**Issue by Chealer** [jquery/jquery.com#114]
The homepage contains:
>Lightweight Footprint

>Only 32kB minified and gzipped. Can also be included as an AMD module

I tried to identify why our intranet took over half a second to load, most of which was not from PHP. There are in the end several reasons, but jQuery took more than I thought. It's hard to say precisely, but just evaluating jQuery UI and jQuery 1.10 on a fairly performant PC (2 years old AMD A8-5600K APU) takes over 10 cs.

Now, I acknowledge that jQuery UI is not "jQuery Core". At 46 ms, I would not call jQuery Core 1.10 heavyweight. Furthermore, http://modernweb.com/2014/03/10/is-jquery-too-big-for-mobile/ suggests I would get much better evaluation times (which might reach under 1 cs) with jQuery 2.1.

On the other hand, I did not realize I was adding such a footprint to the intranet when I included jQuery and jQuery UI by default. And I did think about performance. But - and I reckon I might have been ignorant/careless then - I thought the main impact would be the time taken to fetch jQuery (HTTP). And since our server is local, that was a non-issue. I absolutely did not see script evaluation time.

However, looking at the quote above, I can understand why someone would only think about download time. While download time is indeed much more important on first load (although it has been pointed out that using CDN-s, even the first load may not require downloading jQuery), in the end, evaluation is much more important. If we want to quantify footprint by size, we should give the standard or minified size (probably the latter indeed), not a gzipped size.

While this ticket is strictly about the qualification "gzipped", I mention several related things I noted:
1. The size alone does not make it easy for anyone to get a good idea of the performance impact. Documentation like http://modernweb.com/2014/03/10/is-jquery-too-big-for-mobile/ would be most useful.
2. The description says "Can also be included as an AMD module". I am far from an experienced JavaScript developer, but even after 10 minutes of reading on AMD, I could not figure how this would impact jQuery's footprint. This should surely be removed or changed to link to an explanation.
3. Lightweight Footprint is not a reason why I would use jQuery, unless I am comparing with other general JavaScript libraries. It is only a lesser disadvantage. The homepage should make it more prominent that jQuery is a JavaScript library than that its footprint would be small.

**Comment by dmethvin** [comment]
This is the main jQuery page, it doesn't make sense to fill it with details. If anything that number should go down, the 2.x branch is less than 30K, but I'm fine with leaving it as-is. 

Can you summarize specifically what you think needs to be done here?

**Comment by Chealer** [comment]
Dave, I assume you were replying on point 1. I absolutely agree. What I meant is that the "description" of "Lightweight Footprint" is not very convincing. I could not find documentation on footprint. If there is none, there should be a section on footprint, and the description should link to that to back it up.

Specifically, this ticket is simply asking not to just give jQuery's gzipped size on the homepage. I am not asking to remove that size, but to make it no more prominent than an uncompressed size. In practice, I think this means the gzipped size should be removed, since giving 2 sizes seems too verbose for the homepage. Therefore, unless someone has another suggestion, my concrete proposal is to replace the gzipped size with an uncompressed size.

**Comment by kswedberg** [comment]
I don't understand the rationale for giving 2 sizes or simply removing the gzipped size. Who in their right mind would not configure their server to gzip (or deflate, etc.) their js and css assets?

**Comment by dmethvin** [comment]
> Specifically, this ticket is simply asking not to just give jQuery's gzipped size on the homepage. 

That's been a pretty standard way of stating the size of the library since forever, so I wouldn't want to remove it.

**Comment by Chealer** [comment]
kswedberg, numerous enterprise administrators including myself would not (in a LAN context).

The rationale is that a compressed size says little about evaluation time. Unless evaluation performance indications are added, just giving a compressed size suggests that download is the greatest footprint.

**Comment by Chealer** [comment]
dmethvin, why did you close this ticket?

**Comment by dmethvin** [comment]
It sounded like the discussion came down to removing the gzipped size, which I don't think we'd want to do.

**Comment by Chealer** [comment]
dmethvin, I recommended replacing the gzipped size with an uncompressed size (and therefore indeed removing the gzipped size) because I thought that was the optimal solution. But as I said, there are other ways to solve this. As long as an uncompressed size becomes at least as prominent as compressed sizes (I would **recommend** making uncompressed **more** prominent), I will consider this solved.

On second thought, including an uncompressed and a compressed size can be done without being that verbose. We could simply say "Only x minified (y gzipped)". Technically, that is just one more word.

**Comment by dmethvin** [comment]
That would work I think. Can you put together a pull request? I'm not sure which version to use, I would lean towards the latest 2.x at this point I think.

**Comment by Chealer** [comment]
I have never done that, sorry.

I agree that using version 1 seems undesirable. However, if it is still offered, just discussing version 2 could  be misleading. I suggest either:
1. Using just version 2 size but specify that
2. Showing numbers for each version
3. Keep what we have as long as version 1 is maintained, but specify "or less" on sizes

**Comment by mzgol** [comment]
> I agree that using version 1 seems undesirable. However, if it is still offered, just discussing version 2 could be misleading.

Maybe we could changing the text by 2-3 months then? We'll then release jQuery 3.0.0 and there will be only one supported line; jQuery 1 & 2 will go away.

**Comment by Chealer** [comment]
I reported the (core) issue in ticket #148.
